sidecars/filesyncer/providers/s3/s3.go

```diff
@@ -34,13 +35,29 @@ func Sync() {
 		util.Mux.Unlock()
 	}
 
+	config, err := util.GetArtifactRepositoryConfig()
+	if err != nil {
+		log.Printf("[error] unable to get artifact repository config")
+		return
+	}
+
+	gcp := config.S3.Endpoint != "s3.amazonaws.com"
+
 	var cmd *exec.Cmd
 	uri := fmt.Sprintf("s3://%v/%v", util.Bucket, util.Prefix)
 	if util.Action == util.ActionDownload {
-		cmd = util.Command("aws", "s3", "sync", "--delete", uri, util.Path)
+		if gcp {
+			cmd = util.Command("aws", "s3", "--endpoint-url https://storage.googleapis.com", "sync", "--delete", util.Path, uri)
+		} else {
+			cmd = util.Command("aws", "s3", "sync", "--delete", uri, util.Path)
+		}
 	}
 	if util.Action == util.ActionUpload  {
-		cmd = util.Command("aws", "s3", "sync", "--delete", util.Path, uri)
+		if gcp {
```

Comment:
Would it make sense to make this so that it's not GCS specific? For example for Minio?